src/CommunityToolkit.Mvvm.SourceGenerators/ComponentModel/ObservablePropertyGenerator.Execute.cs

```diff
@@ -668,6 +675,46 @@ private static bool IsOldPropertyValueDirectlyReferenced(IFieldSymbol fieldSymbo
             return false;
         }
 
+        /// <summary>
+        /// Gets the nullability info on the generated property
+        /// </summary>
+        /// <param name="fieldSymbol">The input <see cref="IFieldSymbol"/> instance to process.</param>
+        /// <param name="semanticModel">The <see cref="SemanticModel"/> instance for the current run.</param>
+        /// <param name="isReferenceTypeOrUnconstraindTypeParameter">Whether the property type supports nullability.</param>
+        /// <param name="includeMemberNotNullOnSetAccessor">Whether <see cref="MemberNotNullAttribute"/> should be used on the setter.</param>
+        /// <returns></returns>
+        private static void GetNullabilityInfo(
+            IFieldSymbol fieldSymbol,
+            SemanticModel semanticModel,
+            out bool isReferenceTypeOrUnconstraindTypeParameter,
+            out bool includeMemberNotNullOnSetAccessor)
+        {
+            // We're using IsValueType here and not IsReferenceType to also conver unconstrained type parameter cases.
+            // This will cover both reference types as well T when the constraints are not struct or unmanaged.
+            isReferenceTypeOrUnconstraindTypeParameter = !fieldSymbol.Type.IsValueType;
+
+            // This is used to avoid nullability warnings when setting the property from a constructor, in case the field
+            // was marked as not nullable. Nullability annotations are assumed to always be enabled to make the logic simpler.
+            // Consider this example:
+            //
+            // partial class MyViewModel : ObservableObject
+            // {
+            //    public MyViewModel()
+            //    {
+            //        Name = "Bob";
+            //    }
+            //
+            //    [ObservableProperty]
+            //    private string name;
+            // }
+            //
+            // The [MemberNotNull] attribute is needed on the setter for the generated Name property so that when Name
+            // is set, the compiler can determine that the name backing field is also being set (to a non null value).
+            includeMemberNotNullOnSetAccessor =
+                fieldSymbol.Type is { NullableAnnotation: not NullableAnnotation.Annotated, IsValueType: false } &&
```

Comment:
Stylistic suggestion. `IsValueType` was already checked above, so reused `isReferenceTypeOrUnconstraindTypeParameter`. Then the pattern ended up with a single property, so checked it directly. You can definitely ignore this if you like the original code more.

```suggestion
            includeMemberNotNullOnSetAccessor =
                isReferenceTypeOrUnconstraindTypeParameter &&
                fieldSymbol.Type.NullableAnnotation != NullableAnnotation.Annotated &&
```

Reply:
Makes sense! I've integrated the changes in the new commit with the other tweaks too. Thank you! 🙂